docs/upgrade/UPGRADE-unreleased.md

```diff
@@ -19,4 +19,89 @@ There you can find links to upgrade notes for other versions too.
+                - '#Undefined variable: \$undefined#'
+    +           #ignore annotations in generated code#
+    +           -
+    +               message: '#(PHPDoc tag @(param|return) has invalid value .+ expected TOKEN_IDENTIFIER at offset \d+)#'
+    +               path: %currentWorkingDirectory%/tests/ShopBundle/Test/Codeception/_generated/AcceptanceTesterActions.php
+    +           -
+    +               message: '#(PHPDoc tag @throws with type .+ is not subtype of Throwable)#'
+    +               path: %currentWorkingDirectory%/tests/ShopBundle/Test/Codeception/_generated/AcceptanceTesterActions.php
+    +           -
+    +               message: '#Call to an undefined method Symfony\\Component\\Config\\Definition\\Builder\\NodeParentInterface::end\(\)#'
+    +               path: %currentWorkingDirectory%/src/Shopsys/ShopBundle/DependencyInjection/Configuration.php
+    +           - '#Method Doctrine\\Common\\Persistence\\ObjectManager::flush\(\) invoked with 1 parameter, 0 required\.#'
+    +           -
+    +               message: '#(Property Shopsys\\.+::\$.+ \(Shopsys\\.+\) does not accept object\.)#'
+    +               path: %currentWorkingDirectory%/src/Shopsys/ShopBundle/DataFixtures/*
+    +           -
+    +               message: '#Method Shopsys\\ShopBundle\\DataFixtures\\ProductDataFixtureReferenceInjector::.+\(\) should return array<.+> but returns array<string, object>\.#'
+    +               path: %currentWorkingDirectory%/src/Shopsys/ShopBundle/DataFixtures/ProductDataFixtureReferenceInjector.php
+    +           -
+    +               message: '#(Property (Shopsys|Tests)\\.+::\$.+ \(.+\) does not accept object\.)#'
+    +               path: %currentWorkingDirectory%/tests/ShopBundle/*
+    +           -
+    +               message: '#(Method .+::.+\(\) should return .+ but returns (object|Codeception\\Module).)#'
+    +               path: %currentWorkingDirectory%/tests/ShopBundle/*
+    +           -
+    +               message: '#Array \(array<.+>\) does not accept object\.#'
+    +               path: %currentWorkingDirectory%/tests/ShopBundle/Functional/Model/Product/ProductOnCurrentDomainFacadeCountDataTest.php
+    +   includes:
+    +       - vendor/phpstan/phpstan-doctrine/extension.neon
+    +       - vendor/phpstan/phpstan-phpunit/extension.neon
+    ```
+    - after upgrading the version you can expect that PHPStan will start reporting errors that can be caused by:
+        - wrong annotation
+        ```diff
+        -   * @return string[]
+        +   * @return \Shopsys\FrameworkBundle\Model\Pricing\Vat\Vat[]                
+        ```
+        - missing annotation
+        ```diff
+            $treeBuilder = new TreeBuilder();
+        +   /** @var \Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition $rootNode */
+            $rootNode = $treeBuilder->root('shopsys_shop');
+        ```
+    - you can get inspiration on how to fix them in [this commit.]()
```

Comment:
dont forget to update this with URL, maybe this could be part of the previous `-`